src/ortc.ts

```diff
@@ -1043,53 +1044,74 @@ export function getConsumerRtpParameters(
 
-	// If any of the consumableParams.encodings has scalabilityMode, process it
-	// (assume all encodings have the same value).
-	const encodingWithScalabilityMode =
-		consumableParams.encodings!.find((encoding) => encoding.scalabilityMode);
+		// If any of the consumableParams.encodings has scalabilityMode, process it
+		// (assume all encodings have the same value).
+		const encodingWithScalabilityMode =
+			consumableParams.encodings!.find((encoding) => encoding.scalabilityMode);
 
-	let scalabilityMode = encodingWithScalabilityMode
-		? encodingWithScalabilityMode.scalabilityMode
-		: undefined;
+		let scalabilityMode = encodingWithScalabilityMode
+			? encodingWithScalabilityMode.scalabilityMode
+			: undefined;
 
-	// If there is simulast, mangle spatial layers in scalabilityMode.
-	if (consumableParams.encodings!.length > 1)
-	{
-		const { temporalLayers } = parseScalabilityMode(scalabilityMode);
+		// If there is simulast, mangle spatial layers in scalabilityMode.
+		if (consumableParams.encodings!.length > 1)
+		{
+			const { temporalLayers } = parseScalabilityMode(scalabilityMode);
 
-		scalabilityMode = `S${consumableParams.encodings!.length}T${temporalLayers}`;
-	}
+			scalabilityMode = `S${consumableParams.encodings!.length}T${temporalLayers}`;
+		}
 
-	if (scalabilityMode)
-		consumerEncoding.scalabilityMode = scalabilityMode;
+		if (scalabilityMode)
+			consumerEncoding.scalabilityMode = scalabilityMode;
 
-	// Use the maximum maxBitrate in any encoding and honor it in the Consumer's
-	// encoding.
-	const maxEncodingMaxBitrate =
-		consumableParams.encodings!.reduce((maxBitrate, encoding) => (
-			encoding.maxBitrate && encoding.maxBitrate > maxBitrate
-				? encoding.maxBitrate
-				: maxBitrate
-		), 0);
+		// Use the maximum maxBitrate in any encoding and honor it in the Consumer's
+		// encoding.
+		const maxEncodingMaxBitrate =
+			consumableParams.encodings!.reduce((maxBitrate, encoding) => (
+				encoding.maxBitrate && encoding.maxBitrate > maxBitrate
+					? encoding.maxBitrate
+					: maxBitrate
+			), 0);
 
-	if (maxEncodingMaxBitrate)
-	{
-		consumerEncoding.maxBitrate = maxEncodingMaxBitrate;
+		if (maxEncodingMaxBitrate)
+		{
+			consumerEncoding.maxBitrate = maxEncodingMaxBitrate;
+		}
+
+		// Set a single encoding for the Consumer.
+		consumerParams.encodings!.push(consumerEncoding);
 	}
+	else
+	{
+		const consumableEncodings =
+			utils.clone(consumableParams.encodings) as RtpEncodingParameters[];
+		const baseSsrc = utils.generateRandomNumber();
+		const baseRtxSsrc = utils.generateRandomNumber();
+
+		for (let i = 0; i < consumableEncodings.length; ++i)
+		{
+			const encoding = consumableEncodings[i];
+
+			encoding.ssrc = baseSsrc + i;
 
-	// Set a single encoding for the Consumer.
-	consumerParams.encodings!.push(consumerEncoding);
+			if (rtxSupported)
+				encoding.rtx = { ssrc: baseRtxSsrc + i };
+			else
+				delete encoding.rtx;
 
-	// Copy verbatim.
-	consumerParams.rtcp = consumableParams.rtcp;
```

Comment:
Was RTCP removed intentionally here?

Reply:
It was already copied in the object created above at the beginning of the method.

Reply:
Got it.